### DISCUSSION TITLE
Port Forwarding and Transit

### DISCUSSION THREAD

**Discussion by tengkuizdihar** [PurpleI2P/i2pd#1963] [Q&A] [answered]
Hello, new guy here, wants to know about Port Forwarding and Transit.

**Question**:
1. What does port forwarding does and where can I find/specify the port in the config?
2. Right now I have firewalled for "network status" in localhost:7070, but I'm still getting transit tunnels in it. Why is that? Not that I mind, I want to participate but it's not clear enough right now on why it's like that.

Thank you, in advance :pray:

**Comment by Vort** [comment, marked as answer]
1. What port forwarding are you talking about? If you mean forwards from localhost address to i2p address, then look into `tunnels.conf`.
2. There are two mechanisms: (a) Your node makes outgoing transport connections to remote node. Then remote node have possibility to build transit tunnels through your node. (b) Your node can accept incoming SSU2 connections with the help of "introducers" ([UDP hole punching](https://en.wikipedia.org/wiki/UDP_hole_punching)).

**Reply by tengkuizdihar** [reply]
Thanks, your 2. answers my questions.

**Reply by slrslr** [reply]
> UDP hole punching

Isn't bittorrent doing similar or the same thing and still fails to interconnect two passive peers (that are behind NAT and do not have forwarded ports)? So I2P can fail same way and port forwarding is still required to interconnect such I2P peers? If that is so, maybe it should be mentioned in the .conf file or somewhere.

**Reply by Vort** [reply]
@slrslr, ability to connect two passive peers depends on type of their NATs - it is possible only with some combinations (I don't remember details right now). Of course, forwarded port is better, but it is rarely possible to do forwarding and methods are very specific to ISP and hardware user have.
There are similarities with bittorrent, but there are also differences: without bittorrent connection there will be no download in case of rare file. With I2P, there always will be peers with public IP, which can be used to connect peers indirectly.
